plugins/flashrom/fu-plugin-flashrom.c

```diff
@@ -144,17 +158,30 @@ fu_plugin_update_prepare (FuPlugin *plugin,
 	if (!fu_common_mkdir_parent (firmware_orig, error))
 		return FALSE;
 	if (!g_file_test (firmware_orig, G_FILE_TEST_EXISTS)) {
-		const gchar *argv[] = {
-			data->flashrom_fn,
-			"--programmer", "internal:laptop=force_I_want_a_brick",
-			"--read", firmware_orig,
-			"--verbose", NULL };
-		if (!fu_common_spawn_sync ((const gchar * const *) argv,
-					   fu_plugin_flashrom_read_cb, device,
-					   NULL, error)) {
-			g_prefix_error (error, "failed to get original firmware: ");
+		if (flashrom_programmer_init(&flashprog, "internal", NULL)) {
+			g_error ("Error: Programmer initialization failed.");
+			return FALSE;
+		}
+
+		if (flashrom_flash_probe(&flashctx, flashprog, NULL)) {
+			g_error ("Error: Flash probe failed.");
+			return FALSE;
+		}
+		const size_t flash_size = flashrom_flash_getsize(flashctx);
+		uint8_t *newcontents = malloc(flash_size);
+		if (!newcontents) {
```

Comment:
pointers we compare explicitly to NULL, or != NULL. In GLib code you don't need to check for malloc failures as OOM means the g_malloc will fail and crash the program.